internal/broker/broker.go

```diff
@@ -57,8 +57,8 @@ type userConfig struct {
 type Broker struct {
 	cfg Config
 
-	providerInfo providers.ProviderInfoer
-	oidcCfg      oidc.Config
+	provider providers.Provider
```

Comment:
This is tricky... We would then have an oidcProvider and a Provider, both doing different things... I'm not in love with the name `ProviderInfoer` also, but I think naming the interface only as `Provider` could be annoying (@didrocks, IIRC, we tried that name before, right?)